### PR TITLE
Create GitHub action to publish npm package

### PR DESCRIPTION
**Pull request by lym953**
### Motivation
Right now, when someone wants to publish a release for this repo by running `publish_prod.sh`, they need to reach out to someone to request an NPM token for themselves, one token for each person. This is not scalable. As @astuyve and @yoannmoinet suggested, let's set up a GitHub action to release the npm package.

### Previous work
1. I granted @yoannmoinet admin permission of this repo
2. @yoannmoinet added an NPM token as a secret of this repo
3. I revoked admin permission from @yoannmoinet

### What does this PR do?
Set up a GitHub Action to release the npm package following this internal wiki: [Publish a package on NPM](https://datadoghq.atlassian.net/wiki/spaces/~194742439/pages/4180476110/Publish+a+package+on+NPM#)

A message will be sent to Slack channel `serverless-onboarding-and-enablement-ops` with the success/failure result.

### Testing Guidelines
I'm not sure how to test it without actually creating a release. I'd test it in production. i.e.
1. merge this commit
2. run `publish_prod.sh`, which creates a release and will ideally trigger the GitHub Action

To test Slack messaging, I added this GitHub action, which successfully sent a message.
```
name: Yiming's Slack test
on: [push]
jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v3

  notify:
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Send success message to Slack
        env:
          SLACK_CHANNEL: "#serverless-onboarding-and-enablement-ops"
          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
        if: success()
        run: |
          set -x
          OPS_MESSAGE=":gh-check-passed: Yiming is testing Slack bot!"
          curl -H "Content-type: application/json" -X POST "$SLACK_WEBHOOK" -d '{
            "channel": "'"$SLACK_CHANNEL"'",
            "text": "'"$OPS_MESSAGE"'"
          }'
```

image

### Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: scripts/publish_prod.sh
Comment (lym953) [reply scripts/publish_prod.sh:100]: Sure, will add the message and update confluence.

File: .github/workflows/publish.yml
Comment (lym953) [reply .github/workflows/publish.yml:18]: I think the previous `--new-version` is redundant. `--new-version` updates the version number in `package.json` before publishing to npm, but there is already a command: `yarn version --new-version "$VERSION"` in `publish_prod.sh`, which does the same thing. I think the GitHub Action can just read the updated version number from `package.json`.

File: scripts/publish_prod.sh
Comment (lym953) [inline scripts/publish_prod.sh:98]: Updated this message.

File: scripts/publish_prod.sh
Comment (purple4reina) [reply scripts/publish_prod.sh:100]: Oh sorry, I didn't realize the gh action runs automatically. I've got a couple of concerns.

First, we used to do a `yarn login` but now we do not. What impact will that have?

Second, for who are unaware that creating the release in gh triggers a gh action to publish the release to npm. This gh action could fail and we wouldn't know about it. It would be ideal to get some kind of alert on the success/failure of this job. Either I think we should make sure that people go watch the gh action until it completes, or we should post a comment to slack with the final result.

Comment (lym953) [reply scripts/publish_prod.sh:100]: > First, we used to do a yarn login but now we do not. What impact will that have?

`yarn login` prompts the user to log in using username/email and password. Since we are moving to using NPM token for authentication, `yarn login` is no longer needed.

> Second, ...

Sounds good. I'm going to create a Slack App so the GitHub Action can send messages to `serverless-onboarding-and-enablement-ops`. I'm waiting for Datadog's approval on adding this App.

**Comment by lym953** [comment]
Added Slack messaging for NPM publish result

**Review by purple4reina — APPROVED** [review]

**Pull request merged**
